### ISSUE TITLE
DeprecationWarning from `spacy_legacy`

### ISSUE THREAD

**Issue by gautierdag** [allenai/scispacy#350]
Hi there,
I recently upgraded to spacy 3 and scispacy 0.4, but I am now getting a warning whenever I use the small `scispacy` model (I have not tried any other model). 

I am getting a `DeprecationWarning` on a fresh install in python 3.8 with the latest version of `scispacy` and `en_core_sci_sm`.

### Steps to reproduce:
`pip install scispacy`
`pip install https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.4.0/en_core_sci_sm-0.4.0.tar.gz`

```python
import spacy
nlp = spacy.load("en_core_sci_sm")

import warnings
warnings.filterwarnings("error")
nlp("Hello World")
```

Any input to the `nlp` model triggers the same warning:

```
/opt/miniconda3/envs/clean/lib/python3.8/site-packages/spacy_legacy/layers/staticvectors_v1.py in forward(model, docs, is_train)
     43     )
     44     try:
---> 45         vectors_data = model.ops.gemm(model.ops.as_contig(V[rows]), W, trans2=True)
     46     except ValueError:
     47         raise RuntimeError(Errors.E896)

DeprecationWarning: Out of bound index found. 
This was previously ignored when the indexing result contained no elements. 
In the future the index error will be raised. 
This error occurs either due to an empty slice, or if an array has zero elements even before indexing.
(Use `warnings.simplefilter('error')` to turn this DeprecationWarning into an error and get more details on the invalid index.)
```

Any ideas as to how to resolve this without manually ignoring the warning?

**Comment by danielkingai2** [comment]
Could you try installing everything clean in a new environment?

**Comment by gautierdag** [comment]
Yeah sure I can do that, but the deprecation warning is being triggered by the `scispacy` model. If I use a normal core spacy model like `en_core_web_sm` then I don't get a warning / exception.

**Comment by danielkingai2** [comment]
Oh huh. Ok, I might have to look into it further then

**Comment by adrianeboyd** [comment]
Hi, it looks like `en_core_sci_sm` has `include_static_vectors = true` even though there are no vectors, which leads to this deprecation warning in `numpy>=1.20.0`.

**Comment by gautierdag** [comment]
Ok great! Thank you @adrianeboyd - I'll close the issue on the spaCy side.

**Comment by danielkingai2** [comment]
@adrianeboyd do you know if I can modify the model file easily without retraining? or is full retraining with the fixed config necessary?

**Comment by adrianeboyd** [comment]
Hmm, this is one of those cases where I hesitate to describe a workaround, even though I think it's technically possible (you'd have to replace `MultiHashEmbed` and `StaticVectors` with custom code).

But I really wouldn't recommend it, especially for a widely distributed model, so my official answer is: no, I would retrain the model with the corrected config.
